admin/provisioner/kubernetes.go

```diff
@@ -250,8 +251,18 @@ func (p *KubernetesProvisioner) AwaitReady(ctx context.Context, provisionID stri
 		return sts.Status.AvailableReplicas > 0 && sts.Status.AvailableReplicas == sts.Status.Replicas, nil
 	})
 	if err != nil {
-		err2 := p.Deprovision(ctx, provisionID)
-		return multierr.Combine(err, err2)
+		return err
+	}
+
+	// As a final step we make sure the runtime can be reached, we retry on failure, to account for potential small delays in network config propagation
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 3
+	retryClient.RetryWaitMin = 1 * time.Second
+	retryClient.RetryWaitMax = 3 * time.Second
+	retryClient.Logger = nil // Disable inbuilt logger
+	_, err = retryClient.Get(p.getHost(provisionID) + "/v1/ping")
```

Comment:
Use `url.JoinPath` to avoid concerns about whether the host has a trailing slash or not

Reply:
Done